src/unix/newlib/mod.rs

```diff
@@ -87,16 +87,27 @@ s! {
     }
 
     pub struct in_addr {
-            pub s_addr: ::in_addr_t,
+        pub s_addr: ::in_addr_t,
     }
 
     pub struct hostent {
-            pub h_name: *mut ::c_char,
-            pub h_aliases: *mut *mut ::c_char,
-            pub h_addrtype: ::c_int,
-            pub h_length: ::c_int,
-            pub h_addr_list: *mut *mut ::c_char,
-            pub h_addr: *mut ::c_char,
+        pub h_name: *mut ::c_char,
+        pub h_aliases: *mut *mut ::c_char,
+
+        #[cfg(not(target_os = "horizon"))]
+        pub h_addrtype: ::c_int,
+        #[cfg(target_os = "horizon")]
+        pub h_addrtype: u16,
+
+        #[cfg(not(target_os = "horizon"))]
+        pub h_length: ::c_int,
+        #[cfg(target_os = "horizon")]
+        pub h_length: u16,
+
+        pub h_addr_list: *mut *mut ::c_char,
+
+        #[cfg(not(target_os = "horizon"))]
+        pub h_addr: *mut ::c_char,
```

Comment:
Could you instead put this struct in a `cfg_if!` with the current definition under `if #[cfg(not(target_os = "horizon"))] { ... }`, then put the new definition in `horizon/mod.rs`? It's a bit tough to read this way.

Reply:
I thought not to do that originally because there is no way to use `cfg_if!` within `s!`, but I just separated the struct from the rest and made a new definition in `horizon/mod.rs`.